cli/flox-rust-sdk/src/providers/publish.rs

```diff
@@ -167,6 +287,94 @@ impl NixCopyCache {
             Err(PublishError::CacheUploadError(stderr.to_string()))
         }
     }
+
+    /// Constructs a `nix path-info` command that will get the NAR info for a
+    /// store path from the specified store, including the optional information
+    /// about the closure size of the store path.
+    fn nar_info_cmd(store_url: &Url, store_path: &str) -> Command {
+        let mut cmd = nix_base_command();
+        cmd.stdout(Stdio::piped());
+        cmd.stderr(Stdio::piped());
+        cmd.args([
+            "path-info",
+            "--store",
+            store_url.as_str(),
+            "--closure-size",
+            "--json",
+            store_path,
+        ]);
```

Comment:
(from the ticket)

> To note is that the format of the narinfo coming back from nix path-info differs between older and newer versions of nix. This must be considered and data fit into the current API model. The update-catalog code currently does this and can be used as a reference.

I assume we don't need to care about this because we control the version of `nix`?

Reply:
I address that by just taking the JSON here and pulling out the relevant fields in `ClientSideCatalogStoreConfig::extract_catalog_format_nar_info`.

Reply:
Now that I'm serializing into the `NarInfo` struct this method isn't necessary. The catalog server format appears to just be the latest format, and ignores unknown fields e.g. the `path` and `valid` fields in the V1 NAR info format that we drop anyway.

Also, there will be a bug until the `compresssize` -> `compression` typo is fixed on the catalog server side. For context, the `compression` field of the NAR info contains the compression algorithm used e.g. `zstd`, and the `compresssize` field doesn't seem to exist.